[BUG] test_delta_delete_entire_table failed in databricks 10.4 runtime

**Describe the bug**
failed in internal rapids_databricks_nightly-pre_release-github, run ID: 284

cases list
```

[2023-04-14T08:54:58.420Z] FAILED ../../src/main/python/delta_lake_delete_test.py::test_delta_delete_entire_table[None-True][IGNORE_ORDER, ALLOW_NON_GPU(DeserializeToObjectExec,ShuffleExchangeExec,FileSourceScanExec,FilterExec,MapPartitionsExec,MapElementsExec,ObjectHashAggregateExec,ProjectExec,SerializeFromObjectExec,SortExec)] - AssertionError: GPU and CPU int values are different at [0, 'num_affected_r...

[2023-04-14T08:54:58.420Z] FAILED ../../src/main/python/delta_lake_delete_test.py::test_delta_delete_entire_table[None-False][IGNORE_ORDER, ALLOW_NON_GPU(DeserializeToObjectExec,ShuffleExchangeExec,FileSourceScanExec,FilterExec,MapPartitionsExec,MapElementsExec,ObjectHashAggregateExec,ProjectExec,SerializeFromObjectExec,SortExec)] - AssertionError: GPU and CPU int values are different at [0, 'num_affected_r...

[2023-04-14T08:54:58.420Z] FAILED ../../src/main/python/delta_lake_delete_test.py::test_delta_delete_entire_table[['a']-True][IGNORE_ORDER, ALLOW_NON_GPU(DeserializeToObjectExec,ShuffleExchangeExec,FileSourceScanExec,FilterExec,MapPartitionsExec,MapElementsExec,ObjectHashAggregateExec,ProjectExec,SerializeFromObjectExec,SortExec)] - AssertionError: GPU and CPU int values are different at [0, 'num_affected_r...

[2023-04-14T08:54:58.420Z] FAILED ../../src/main/python/delta_lake_delete_test.py::test_delta_delete_entire_table[['a']-False][IGNORE_ORDER, ALLOW_NON_GPU(DeserializeToObjectExec,ShuffleExchangeExec,FileSourceScanExec,FilterExec,MapPartitionsExec,MapElementsExec,ObjectHashAggregateExec,ProjectExec,SerializeFromObjectExec,SortExec)] - AssertionError: GPU and CPU int values are different at [0, 'num_affected_r...

[2023-04-14T08:54:58.420Z] FAILED ../../src/main/python/delta_lake_delete_test.py::test_delta_delete_partitions[['a']-True][INJECT_OOM, IGNORE_ORDER, ALLOW_NON_GPU(DeserializeToObjectExec,ShuffleExchangeExec,FileSourceScanExec,FilterExec,MapPartitionsExec,MapElementsExec,ObjectHashAggregateExec,ProjectExec,SerializeFromObjectExec,SortExec)] - AssertionError: GPU and CPU int values are different at [0, 'num_affected_r...

[2023-04-14T08:54:58.420Z] FAILED ../../src/main/python/delta_lake_delete_test.py::test_delta_delete_partitions[['a']-False][IGNORE_ORDER, ALLOW_NON_GPU(DeserializeToObjectExec,ShuffleExchangeExec,FileSourceScanExec,FilterExec,MapPartitionsExec,MapElementsExec,ObjectHashAggregateExec,ProjectExec,SerializeFromObjectExec,SortExec)] - AssertionError: GPU and CPU int values are different at [0, 'num_affected_r...

[2023-04-14T08:54:58.420Z] FAILED ../../src/main/python/delta_lake_delete_test.py::test_delta_delete_partitions[['a', 'b']-True][INJECT_OOM, IGNORE_ORDER, ALLOW_NON_GPU(DeserializeToObjectExec,ShuffleExchangeExec,FileSourceScanExec,FilterExec,MapPartitionsExec,MapElementsExec,ObjectHashAggregateExec,ProjectExec,SerializeFromObjectExec,SortExec)] - AssertionError: GPU and CPU int values are different at [0, 'num_affected_r...

[2023-04-14T08:54:58.420Z] FAILED ../../src/main/python/delta_lake_delete_test.py::test_delta_delete_partitions[['a', 'b']-False][IGNORE_ORDER, ALLOW_NON_GPU(DeserializeToObjectExec,ShuffleExchangeExec,FileSourceScanExec,FilterExec,MapPartitionsExec,MapElementsExec,ObjectHashAggregateExec,ProjectExec,SerializeFromObjectExec,SortExec)] - AssertionError: GPU and CPU int values are different at [0, 'num_affected_r...
```

failed mismatched cpu and gpu num_affected_rows,
```
[2023-04-14T08:54:58.419Z] cpu = 395, gpu = -1

[2023-04-14T08:54:58.419Z] float_check = <function get_float_check.<locals>.<lambda> at 0x7f801d04b940>

[2023-04-14T08:54:58.419Z] path = [0, 'num_affected_rows']

[2023-04-14T08:54:58.419Z] 

[2023-04-14T08:54:58.419Z]     def _assert_equal(cpu, gpu, float_check, path):

[2023-04-14T08:54:58.419Z]         t = type(cpu)

[2023-04-14T08:54:58.419Z]         if (t is Row):

[2023-04-14T08:54:58.419Z]             assert len(cpu) == len(gpu), "CPU and GPU row have different lengths at {} CPU: {} GPU: {}".format(path, len(cpu), len(gpu))

[2023-04-14T08:54:58.419Z]             if hasattr(cpu, "__fields__") and hasattr(gpu, "__fields__"):

[2023-04-14T08:54:58.419Z]                 assert cpu.__fields__ == gpu.__fields__, "CPU and GPU row have different fields at {} CPU: {} GPU: {}".format(path, cpu.__fields__, gpu.__fields__)

[2023-04-14T08:54:58.419Z]                 for field in cpu.__fields__:

[2023-04-14T08:54:58.419Z]                     _assert_equal(cpu[field], gpu[field], float_check, path + [field])

[2023-04-14T08:54:58.419Z]             else:

[2023-04-14T08:54:58.419Z]                 for index in range(len(cpu)):

[2023-04-14T08:54:58.419Z]                     _assert_equal(cpu[index], gpu[index], float_check, path + [index])

[2023-04-14T08:54:58.419Z]         elif (t is list):

[2023-04-14T08:54:58.419Z]             assert len(cpu) == len(gpu), "CPU and GPU list have different lengths at {} CPU: {} GPU: {}".format(path, len(cpu), len(gpu))

[2023-04-14T08:54:58.419Z]             for index in range(len(cpu)):

[2023-04-14T08:54:58.419Z]                 _assert_equal(cpu[index], gpu[index], float_check, path + [index])

[2023-04-14T08:54:58.419Z]         elif (t is tuple):

[2023-04-14T08:54:58.419Z]             assert len(cpu) == len(gpu), "CPU and GPU list have different lengths at {} CPU: {} GPU: {}".format(path, len(cpu), len(gpu))

[2023-04-14T08:54:58.419Z]             for index in range(len(cpu)):

[2023-04-14T08:54:58.419Z]                 _assert_equal(cpu[index], gpu[index], float_check, path + [index])

[2023-04-14T08:54:58.419Z]         elif (t is pytypes.GeneratorType):

[2023-04-14T08:54:58.419Z]             index = 0

[2023-04-14T08:54:58.419Z]             # generator has no zip :( so we have to do this the hard way

[2023-04-14T08:54:58.419Z]             done = False

[2023-04-14T08:54:58.419Z]             while not done:

[2023-04-14T08:54:58.419Z]                 sub_cpu = None

[2023-04-14T08:54:58.419Z]                 sub_gpu = None

[2023-04-14T08:54:58.419Z]                 try:

[2023-04-14T08:54:58.419Z]                     sub_cpu = next(cpu)

[2023-04-14T08:54:58.419Z]                 except StopIteration:

[2023-04-14T08:54:58.419Z]                     done = True

[2023-04-14T08:54:58.419Z]     

[2023-04-14T08:54:58.419Z]                 try:

[2023-04-14T08:54:58.419Z]                     sub_gpu = next(gpu)

[2023-04-14T08:54:58.419Z]                 except StopIteration:

[2023-04-14T08:54:58.419Z]                     done = True

[2023-04-14T08:54:58.419Z]     

[2023-04-14T08:54:58.419Z]                 if done:

[2023-04-14T08:54:58.419Z]                     assert sub_cpu == sub_gpu and sub_cpu == None, "CPU and GPU generators have different lengths at {}".format(path)

[2023-04-14T08:54:58.419Z]                 else:

[2023-04-14T08:54:58.419Z]                     _assert_equal(sub_cpu, sub_gpu, float_check, path + [index])

[2023-04-14T08:54:58.419Z]     

[2023-04-14T08:54:58.419Z]                 index = index + 1

[2023-04-14T08:54:58.419Z]         elif (t is dict):

[2023-04-14T08:54:58.419Z]             # The order of key/values is not guaranteed in python dicts, nor are they guaranteed by Spark

[2023-04-14T08:54:58.419Z]             # so sort the items to do our best with ignoring the order of dicts

[2023-04-14T08:54:58.419Z]             cpu_items = list(cpu.items()).sort(key=_RowCmp)

[2023-04-14T08:54:58.420Z]             gpu_items = list(gpu.items()).sort(key=_RowCmp)

[2023-04-14T08:54:58.420Z]             _assert_equal(cpu_items, gpu_items, float_check, path + ["map"])

[2023-04-14T08:54:58.420Z]         elif (t is int):

[2023-04-14T08:54:58.420Z] >           assert cpu == gpu, "GPU and CPU int values are different at {}".format(path)

[2023-04-14T08:54:58.420Z] E           AssertionError: GPU and CPU int values are different at [0, 'num_affected_rows']
```

**Steps/Code to reproduce bug**
command used in CI,
```
/databricks/spark/bin/spark-submit --jars /home/ubuntu/spark-rapids/dist/target/rapids-4-spark_2.12-23.04.0-SNAPSHOT-cuda11.jar,/home/ubuntu/spark-rapids/integration_tests/target/rapids-4-spark-integration-tests_2.12-23.04.0-SNAPSHOT-spark321db.jar,/home/ubuntu/spark-rapids/integration_tests/target/dependency/parquet-hadoop-1.12.0-tests.jar --driver-java-options '-ea -Duser.timezone=UTC -Ddelta.log.cacheSize=10 ' --driver-memory 2g --conf spark.rapids.memory.gpu.allocSize=1536m /home/ubuntu/spark-rapids/integration_tests/runtests.py --rootdir /home/ubuntu/spark-rapids/integration_tests /home/ubuntu/spark-rapids/integration_tests/src/main/python -v -rfE '' --std_input_path=/home/ubuntu/spark-rapids/integration_tests/src/test/resources --color=yes '' --junitxml=TEST-pytest-1681460898312661262.xml --runtime_env=databricks -m delta_lake --delta_lake --test_type=nightly
```

**Expected behavior**
Pass the test

--- a/docs/additional-functionality/delta-lake-support.md
+++ b/docs/additional-functionality/delta-lake-support.md
@@ -125,6 +125,16 @@ spark.rapids.sql.command.DeleteCommandEdge=true on Databricks platforms.
 Deleting data from Delta Lake tables via the SQL `DELETE FROM` statement or via the DeltaTable
 `delete` API is supported.
 
+### num_affected_rows Difference with Databricks
+
+The Delta Lake delete command returns a single row result with a `num_affected_rows` column.
+When entire partition files in the table are deleted, the open source Delta Lake and RAPIDS
+Acclerator implementations of delete can return -1 for `num_affected_rows` since it could be
+expensive to open the files and produce an accurate row count. Databricks changed the behavior
+of delete operations that delete entire partition files to return the actual row count.
+This is only a difference in the statistics of the operation, and the table contents will still
+be accurately deleted with the RAPIDS Accelerator.
+
 ## Update Operations on Delta Lake Tables
 
 Delta Lake update acceleration is experimental and is disabled by default. To enable acceleration

--- a/integration_tests/src/main/python/delta_lake_delete_test.py
+++ b/integration_tests/src/main/python/delta_lake_delete_test.py
@@ -37,7 +37,8 @@ def delta_sql_delete_test(spark_tmp_path, use_cdf, dest_table_func, delete_sql,
 
 def assert_delta_sql_delete_collect(spark_tmp_path, use_cdf, dest_table_func, delete_sql,
                                     partition_columns=None,
-                                    conf=delta_delete_enabled_conf):
+                                    conf=delta_delete_enabled_conf,
+                                    skip_sql_result_check=False):
     def read_data(spark, path):
         read_func = read_delta_path_with_cdf if use_cdf else read_delta_path
         df = read_func(spark, path)
@@ -45,10 +46,11 @@ def assert_delta_sql_delete_collect(spark_tmp_path, use_cdf, dest_table_func, de
     def checker(data_path, do_delete):
         cpu_path = data_path + "/CPU"
         gpu_path = data_path + "/GPU"
-        # compare resulting dataframe from the delete operation (some older Spark versions return empty here)
-        cpu_result = with_cpu_session(lambda spark: do_delete(spark, cpu_path).collect(), conf=conf)
-        gpu_result = with_gpu_session(lambda spark: do_delete(spark, gpu_path).collect(), conf=conf)
-        assert_equal(cpu_result, gpu_result)
+        if not skip_sql_result_check:
+            # compare resulting dataframe from the delete operation (some older Spark versions return empty here)
+            cpu_result = with_cpu_session(lambda spark: do_delete(spark, cpu_path).collect(), conf=conf)
+            gpu_result = with_gpu_session(lambda spark: do_delete(spark, gpu_path).collect(), conf=conf)
+            assert_equal(cpu_result, gpu_result)
         # compare table data results, read both via CPU to make sure GPU write can be read by CPU
         cpu_result = with_cpu_session(lambda spark: read_data(spark, cpu_path).collect(), conf=conf)
         gpu_result = with_cpu_session(lambda spark: read_data(spark, gpu_path).collect(), conf=conf)
@@ -94,8 +96,14 @@ def test_delta_delete_entire_table(spark_tmp_path, use_cdf, partition_columns):
                             SetValuesGen(StringType(), "abcdefg"),
                             string_gen)
     delete_sql = "DELETE FROM delta.`{path}`"
+    # Databricks recently changed how the num_affected_rows is computed
+    # on deletes of entire files, RAPIDS Accelerator has yet to be updated.
+    # https://github.com/NVIDIA/spark-rapids/issues/8123
+    skip_sql_result = is_databricks_runtime()
     assert_delta_sql_delete_collect(spark_tmp_path, use_cdf, generate_dest_data,
-                                    delete_sql, partition_columns)
+                                    delete_sql, partition_columns,
+                                    skip_sql_result_check=skip_sql_result)
+
 @allow_non_gpu(*delta_meta_allow)
 @delta_lake
 @ignore_order
@@ -109,8 +117,13 @@ def test_delta_delete_partitions(spark_tmp_path, use_cdf, partition_columns):
                             SetValuesGen(StringType(), "abcdefg"),
                             string_gen)
     delete_sql = "DELETE FROM delta.`{path}` WHERE a = 3"
+    # Databricks recently changed how the num_affected_rows is computed
+    # on deletes of entire files, RAPIDS Accelerator has yet to be updated.
+    # https://github.com/NVIDIA/spark-rapids/issues/8123
+    skip_sql_result = is_databricks_runtime()
     assert_delta_sql_delete_collect(spark_tmp_path, use_cdf, generate_dest_data,
-                                    delete_sql, partition_columns)
+                                    delete_sql, partition_columns,
+                                    skip_sql_result_check=skip_sql_result)
 
 @allow_non_gpu(*delta_meta_allow)
 @delta_lake

